### PR TITLE
196 mastercard

### PR DESCRIPTION
**Pull request by SteveScott**
Making a pull request so I can comment on it. Still waiting for AWS credentials to back up download to AWS (runner.sh line 25)

### CODE REVIEW

File: recipes/mastercard/create_mastercard.sql [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+INSERT INTO mastercard.zip_to_borough values ('10302','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10304','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10314','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10305','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10306','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10308','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10312','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10309','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10307','SI',5);
+INSERT INTO mastercard.zip_to_borough values ('10310','SI',5);
```
Comment (YuliaZamriy) [inline recipes/mastercard/create_mastercard.sql:262]: We'll need to double check this list once Mastercard sends us correct SI data.

File: recipes/mastercard/create_mastercard.sql [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    tmp.nation_name as nation_name,
+    tmp.zip_code as zip_code,
+    tmp.txn_amt as txn_amt_index,
+    tmp.txn_cnt as txn_cnt_index,
+    tmp.acct_cnt as acct_cnt_index,
+    tmp.avg_ticket as avg_ticket_index,
+    tmp.avg_freq as avg_freq_index,
+    tmp.avg_spend_amt as avg_spend_amt_index,
+    (CASE 
+        WHEN tmp.yoy_txn_amt = 'None' THEN 0.0
```
Comment (YuliaZamriy) [inline recipes/mastercard/create_mastercard.sql:331]: @SteveScott how many "None"s in raw data? We should check with mastercard what these mean.

File: recipes/mastercard/create_mastercard.sql [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        ELSE CAST(tmp.yoy_txn_amt AS double precision)
+    END
+    ) as yoy_txn_amt,
+    (CASE
+        WHEN tmp.yoy_txn_cnt = 'None' THEN 0.0
```
Comment (YuliaZamriy) [inline recipes/mastercard/create_mastercard.sql:336]: @SteveScott how many "None"s in raw data? We should check with mastercard what these mean.

File: recipes/mastercard/runner.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    mkdir -p output
+    chmod 777 input
+    
+    #For testing purposes
+    #cp Geogrids_NYC_Zip_Code_Level_Jan2019_Feb2021.zip input/
+
+    #will download all files from mastercard. Then mastercard will delete after successfull download.
+    scp -v -P 22022  -i /root/.ssh/id_rsa_axway -o "StrictHostKeyChecking=no" newyorkcity@files.mastercard.com:geoinsights/data/fromMC/* /input
+    echo 'file is ' $(find $BASEDIR/input -name "*.zip" ) 
+    unzip -d /input -P $MASTERCARD_PASSWORD $(find $BASEDIR/input -name "*.zip" | head -1 )
```
Comment (YuliaZamriy) [inline recipes/mastercard/runner.sh:21]: @SteveScott is there a way to implement a check that there is only one file? what is MC uploads multiple files?

File: recipes/mastercard/runner.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        cd output
+        psql $RDP_DATA -c "\COPY (
+            SELECT * FROM $NAME.\"$VERSION\"
+            ) TO stdout DELIMITER ',' CSV HEADER;" > mastercard.csv
+        
+        #Write Version info
+        echo "$VERSION" > version.txt
+    )
+    #can to split the CSVs, but makes a mess.
+    #cat output/mastercard.csv | python3 split_csv.py
```
Comment (YuliaZamriy) [inline recipes/mastercard/runner.sh:48]: @SteveScott If we are not splitting the file, we should clean up the comments to keep only the relevant ones.

Comment (YuliaZamriy) [reply recipes/mastercard/runner.sh:48]: Maybe create a copy of runner.sh that splits the file and store it in some other folder along with split_csv.py to keep mastercard recipe directory clean

File: recipes/mastercard/create_mastercard.sql (hunk 1 above)
Comment (YuliaZamriy) [reply recipes/mastercard/create_mastercard.sql:262]: We should implement a check:
- to make sure that the number of rows in the raw and processed data are the same
- if the number of rows is not the same, then most likely there are new zip codes (unlikely, but possible) that we'll need to add to this code (unless they are not NYC).
- what happens if some of this zip codes are dropped?

**Review by YuliaZamriy — APPROVED** [review]
@SteveScott Looks good. I have a few suggestions in the comments.

File: recipes/mastercard/create_mastercard.sql (hunk 1 above)
Comment (SteveScott) [reply recipes/mastercard/create_mastercard.sql:262]: How do you want it to fail if some NYC zip codes are not included in the dataset @objornsson

File: recipes/mastercard/runner.sh (hunk 5 above)
Comment (SteveScott) [reply recipes/mastercard/runner.sh:48]: Now that it is committed in version control that is all the record we need. I can remove it.

**Pull request merged**